node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1149 @@
+
+							for watcher in self.backing_watchers.iter_mut() {
+								let _ = watcher.send(NewBackedCandidate(backed.clone())).await;
+							}
+						}
+					}
+
+					let mut reports = Vec::new();
+
+					for (k, v) in self.table.get_misbehavior().iter() {
+						if !self.reported_misbehavior_for.contains(k) {
+							self.reported_misbehavior_for.insert(*k);
+
+							if let Ok(report) = MisbehaviorReport::try_from((*k, v.clone())) {
+								let message = ProvisionerMessage::ProvisionableData(
+									ProvisionableData::MisbehaviorReport(self.parent, report)
+								);
+
+								reports.push(message);
+							}
+						}
+					}
+
+					for report in reports.drain(..) {
+						self.send_to_provisioner(report).await?
+					}
+				}
+				Err(()) => {
+					return Err(Error::ValidationFailed(ValidationFailed));
+				}
+			}
+		};
+
+		Ok(())
+	}
+
+	async fn process_msg(&mut self, msg: CandidateBackingMessage) -> Result<(), Error> {
+		match msg {
+			CandidateBackingMessage::Second(_, candidate) => {
+				// If the message is a `CandidateBackingMessage::Second`, sign and dispatch a
+				// Seconded statement only if we have not seconded any other candidate and
+				// have not signed a Valid statement for the requested candidate.
+				match self.seconded {
+					// This job has not seconded a candidate yet.
+					None => {
+						let candidate_hash = candidate.hash();
+
+						if !self.issued_validity.contains(&candidate_hash) {
+							if let Ok(()) = self.validate_and_second(candidate).await {
+								self.seconded = Some(candidate_hash);
+							}
+						}
+					}
+					// This job has already seconded a candidate.
+					Some(_) => {
+					},
+				}
+			}
+			CandidateBackingMessage::Statement(_, statement) => {
+				let _ = self.import_statement(statement).await;
+			}
+			CandidateBackingMessage::RegisterBackingWatcher(_, tx) => {
+				self.backing_watchers.push(tx);
+			}
+		}
+
+		Ok(())
+	}
+
+	async fn send_to_provisioner(&mut self, msg: ProvisionerMessage) -> Result<(), Error> {
+		self.tx_from.send(FromJob::Provisioner(msg)).await?;
+
+		Ok(())
+	}
+
+	async fn request_candidate_validation(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+		pov: PoVBlock,
+	) -> Result<(), Error> {
+		let (tx, rx) = oneshot::channel();
+
+		self.tx_from.send(FromJob::CandidateValidation(
+				CandidateValidationMessage::Validate(
+					self.parent,
+					candidate,
+					pov,
+					tx,
+				)
+			)
+		).await?;
+
+		rx.await??;
+
+		Ok(())
+	}
+
+	async fn distribute_signed_statement(&mut self, s: SignedFullStatement) -> Result<(), Error> {
+		let smsg = StatementDistributionMessage::Share(self.parent, s);
+
+		self.tx_from.send(FromJob::StatementDistribution(smsg)).await?;
+
+		Ok(())
+	}
+
+	async fn spawn_validation_work(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+	) -> Result<(), Error> {
+		let (tx, rx) = oneshot::channel();
+
+		self.tx_from.send(FromJob::AvailabilityStoreMessage(
+				AvailabilityStoreMessage::QueryPoV(
```

Comment:
I don't see any reason we would have the PoV in this case.

This gets called for two reasons:
  - We get told by the `CandidateSelection` subsystem to second a candidate. In this case, the `PoV` should be sent alongside the candidate by the collator and the message should probably reflect that. There is nowhere else we could get the PoV.
  - We are trying to validate a block seconded by another validator. In this case, we should be getting the `PoV` from the `PoVDistribution` subsystem.

So the origin of the PoV is highly dependent on the caller, and I'd recommend just passing it as an argument to `spawn_validation_work` to be passed onwards to `request_candidate_validation`. But in that case, `spawn_validation_work` would just be a thin wrapper around `request_candidate_validation`, so they should be merged.

Reply:
It looks like `CandidateValidationMessage::Second` already contains the PoV, fortunately.